src/test_pages/BasePage.ts

```diff
@@ -0,0 +1,33 @@
+import { Locator, Page, expect, test } from '@playwright/test';
+
+export abstract class BasePage{
+    constructor(protected page: Page){
+    }
+
+    protected async validatePageUrl(url: string){
+        await test.step(`Validating that a correct value of URL is ${url}`, async()=>{
+            await expect(this.page).toHaveURL(url)
+        })
+    }
+
+    protected async clickOnElement(element: Locator){
+        await test.step(`Click on ${element}`, async()=>{
+            await element.waitFor({state: "visible"})
```

Comment:
not needed
https://playwright.dev/docs/actionability

Reply:
> not needed https://playwright.dev/docs/actionability

Accept this. I'll change it.